### PR TITLE
[CLOUDTRUST-2764] Add method to reset credential failures (paper card)

### PR DESCRIPTION
**Pull request by fperot74**

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -31,7 +31,8 @@
 
 [[constraint]]
   name = "github.com/cloudtrust/keycloak-client"
-  version = "v2.1.0"
+  #version = "v2.1.0"
```
Comment (fperot74) [inline Gopkg.toml:34]: Set new version before merging

File: api/management/swagger-api_management.yaml [hunk 2]
```diff
@@ -787,6 +787,42 @@ paths:
       responses:
         200:
           description: successful operation
+  /realms/{realm}/users/{userID}/credentials/{credentialID}/unlock:
```
Comment (harture) [inline api/management/swagger-api_management.yaml:790]: reset-failures ?

File: api/management/swagger-api_management.yaml [hunk 3]
```diff
@@ -787,6 +787,42 @@ paths:
...
+        required: true
+        schema:
+          type: string
+      - name: credentialID
+        in: path
+        description: Credential id
+        required: true
+        schema:
+          type: string
+      - name: type
```
Comment (harture) [inline api/management/swagger-api_management.yaml:814]: I don't see the benefit of this parameter, I would suggest to remove it to keep it simple

File: pkg/management/authorization.go [hunk 4]
```diff
@@ -52,6 +52,7 @@ var (
 	MGMTCreateRecoveryCode                  = newAction("MGMT_CreateRecoveryCode", security.ScopeGroup)
 	MGMTGetCredentialsForUser               = newAction("MGMT_GetCredentialsForUser", security.ScopeGroup)
 	MGMTDeleteCredentialsForUser            = newAction("MGMT_DeleteCredentialsForUser", security.ScopeGroup)
+	MGMTUnlockCredentialForUser             = newAction("MGMT_UnlockCredentialForUser", security.ScopeGroup)
```
Comment (harture) [inline pkg/management/authorization.go:55]: ResetFailure ?

File: pkg/management/component.go [hunk 5]
```diff
@@ -932,6 +934,50 @@ func (c *component) DeleteCredentialsForUser(ctx context.Context, realmName stri
 	return err
 }
 
+func (c *component) UnlockCredentialForUser(ctx context.Context, realmName string, userID string, credentialID string, credType string) error {
```
Comment (harture) [inline pkg/management/component.go:937]: Do we really need to be able to provide credType as we can determine it by ourself ?

**Pull request merged**